stdlib/src/main/scala/chisel3/std/tilelink/MessageTypes.scala

```diff
@@ -0,0 +1,104 @@
+package chisel3.std.tilelink
+
+import chisel3._
+import chisel3.internal.firrtl.Width
+
+/**
+  * opcode field of TileLink messages
+  *
+  *  cf. TileLink Spec 1.8.1 Table 13
+  */
+object OpCode {
+  protected[tilelink] val width: Width = 3.W
+
+  val PutFullData:    UInt = 0.U(width)
+  val PutPartialData: UInt = 1.U(width)
+  val ArithmeticData: UInt = 2.U(width)
+  val LogicalData:    UInt = 3.U(width)
+  val Get:            UInt = 4.U(width)
+  val Intent:         UInt = 5.U(width)
+  val AcquireBlock:   UInt = 6.U(width)
+  val AcquirePerm:    UInt = 7.U(width)
+  val ProbePerm:      UInt = 7.U(width)
+  val AccessAck:      UInt = 0.U(width)
+  val AccessAckData:  UInt = 1.U(width)
+  val HintAck:        UInt = 2.U(width)
+  val ProbeAck:       UInt = 4.U(width)
+  val ProbeAckData:   UInt = 5.U(width)
+  val Release:        UInt = 6.U(width)
+  val ReleaseData:    UInt = 7.U(width)
+  val Grant:          UInt = 4.U(width)
+  val GrantData:      UInt = 5.U(width)
+  val ProbeBlock:     UInt = 6.U(width)
+  val ReleaseAck:     UInt = 6.U(width)
+  // GrantAck has no Opcode
+}
+
+/**
+  * param field of TileLink messages
+  *
+  * Allowed values are dependent on opcode of the message.
+  */
+object Param {
+  protected[tilelink] val width: Width = 3.W
+
+  val tieZero: UInt = 0.U(width)
+
+  /** TileLink Spec 1.8.1 Table 23
+    */
+  object Arithmetic {
+    val MIN:  UInt = 0.U(width)
+    val MAX:  UInt = 1.U(width)
+    val MINU: UInt = 2.U(width)
+    val MAXU: UInt = 3.U(width)
+    val ADD:  UInt = 4.U(width)
+  }
+
+  /** TileLink Spec 1.8.1 Table 25
+    */
+  object Logical {
+    val XOR:  UInt = 0.U(width)
+    val OR:   UInt = 1.U(width)
+    val AND:  UInt = 2.U(width)
+    val SWAP: UInt = 3.U(width)
+  }
+
+  /** TileLink Spec 1.8.1 Table 27 Intent
+    */
+  object Intent {
+    val PrefetchRead:  UInt = 0.U(width)
+    val PrefetchWrite: UInt = 1.U(width)
+  }
+
+  /** TileLink Spec 1.8.1 Table 31 Cap
+    */
+  object Cap {
+    val toT: UInt = 0.U(width)
+    val toB: UInt = 1.U(width)
+    val toN: UInt = 2.U(width)
+  }
+
+  /** TileLink Spec 1.8.1 Table 31 Grow
+    */
+  object Grow {
+    val NtoB: UInt = 0.U(width)
+    val NtoT: UInt = 1.U(width)
+    val BtoT: UInt = 2.U(width)
+  }
+
+  /** TileLink Spec 1.8.1 Table 31 Prune
+    */
+  object Prune {
+    val TtoB: UInt = 0.U(width)
+    val TtoN: UInt = 1.U(width)
+    val BtoN: UInt = 2.U(width)
+  }
+
+  /** TileLink Spec 1.8.1 Table 31 Report
+    */
+  object Report {
+    val TtoT: UInt = 3.U(width)
```

Comment:
`TToT`

I am also wondering if there isn't a less abbreviated spelling for this that would be a little easier on the eye and less difficult to distinguish visually.

Reply:
See [tilelink spec](https://sifive.cdn.prismic.io/sifive/7bef6f5c-ed3a-4712-866a-1a2e0c6b7b13_tilelink_spec_1.8.1.pdf) Table 31, I prefer strictly comply to spec

Reply:
Probably we can use `TrunkToTrunk` `TrunkToNone` `BranchToNone` etc? rocket-chip also [use `TtoT`](https://github.com/chipsalliance/rocket-chip/blob/661241021e07650872f28413b8e6fdc9c16355ce/src/main/scala/tilelink/Bundles.scala#L95), though.

Reply:
I strongly suggest that we should strictly comply with spec, if one day SiFive update the spec or opensource spec for community, we can PR it, and then change our implementation.